src/tests/unit/handler.test.js

```diff
@@ -1,14 +1,51 @@
 'use strict'
 
-const { handler } = require('../../handler')
+const handler = require('../../handler')
 const chai = require('chai')
 const expect = chai.expect
 const httpStatus = require('http-status-codes')
+const fs = require('fs')
+const nock = require('nock')
+
+const isValidParameterName = (actualKeys, querySpec) => actualKeys.every(key => querySpec.map(param => param.name).includes(key))
+
+const hasRequiredParameters = (actualKeys, querySpec) => {
+  const required = querySpec.filter(param => param.required === true).map(param => param.name)
+  return required.every(item => actualKeys.includes(item))
+}
+
+const hasValidQueryParameters = (requestBody) => {
+  const querySpec = [{ name: 'query', required: true }, { name: 'year', required: false }, { name: 'start', required: false }]
+  if (requestBody.queryStringKeys === undefined) return true
+  const queryStringKeys = Object.keys(requestBody.queryStringParameters)
+  const valid = isValidParameterName(queryStringKeys, querySpec) && hasRequiredParameters(queryStringKeys, querySpec)
+  console.log(`hasValidQueryParameters(${requestBody}) -> ${valid}`)
+  return valid
+}
```

Comment:
Helper methods should be under the tests. I think having the same **general** codestlyle principles helps when we have a polyglot code base.

Reply:
Helpers moved down to bottom